### ISSUE TITLE
[Desktop] x:Bind local does not work in Custom Control

### ISSUE THREAD

**Issue by kucint** [unoplatform/uno#18404]
### Current behavior

**Issue 1**
binding to local constant works in `Page` but not working in `Control`

```xaml
<Grid
  Width="{x:Bind local:MyControl.MyWidth}"
  Height="{x:Bind local:MyControl.MyHeight}" />
```

Code behind:

```cs
public sealed partial class MyControl : Control
{
    public const double MyWidth = 200;
    public const double MyHeight = 200;
}
```

**Issue 2**

Discrepancy between `WinAppSDK` and `Desktop`: `WinAppSDK` works fine with `private const` while `Desktop` requires `public const`

`WinAppSDK`
```cs
public sealed partial class MyControl : Control
{
    private const double MyWidth = 200;
    private const double MyHeight = 200;
}
```
`Desktop` 
```cs
public sealed partial class MyControl : Control
{
    public const double MyWidth = 200;
    public const double MyHeight = 200;
}
```

### Expected behavior

binding to local constant shall work in `Control`

### How to reproduce it (as minimally and precisely as possible)

MINIMAL REPRO PROJECT: [XBindLocalApp.zip](https://github.com/user-attachments/files/17289606/XBindLocalApp.zip)

STEPS TO REPRODUCE:
run the app on Windows with `WinAppSDK` framework selected: all works fine:
![image](https://github.com/user-attachments/assets/2ec91ade-15df-44c3-896d-c96c4e28894d)

run the app on Windows with `Desktop` framework selected: binding works in `Page` but fails in `Control`
![image](https://github.com/user-attachments/assets/9692dc60-9fb4-446c-bbcc-fce0fad40d9d)

### Workaround

define constant in XAML rather than in C# code behind

### Works on UWP/WinUI

Yes

### NuGet package version(s)

`"Uno.Sdk": "5.4.8"`

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

**Comment by kucint** [comment]
@MartinZikmund: still not working on `"Uno.Sdk": "5.5.32"`

**Comment by MartinZikmund** [comment]
@kucint This was more complex than I originally hoped. I think the PR I opened has the fix, but will need to validate it against all tests.

In the meantime a good way to workaround this could be to define the constants in XAML insted, e.g. in a resource dictionary you can do:

```xml
<x:Double x:Name="MyControlSize">200</x:Double>
```

And then use the resource in the template:

```xml
Width="{StaticResource MyControlSize}"
```

Ultimately the result should be equivalent, even performance-wise (and could actually even be more performant).

**Comment by kucint** [comment]
@MartinZikmund : defining constants in resources is XAML is not an option to me.
What could be an option to me is to add constant values to the resource dictionary from code behind.
Would it be as performant as adding them directly in XAML?

**Comment by MartinZikmund** [comment]
@kucint hopefully my PR will avoid the need for this, but yes, you could set the resources early on app startup from C#, it would have the same end result, while keeping the consts in C#

**Comment by kucint** [comment]
I upgraded my test app to the newest dotnet and newest "Uno.Sdk": "5.5.43": [XBindLocalApp.zip](https://github.com/user-attachments/files/17777464/XBindLocalApp.zip)
and it is still not working.

**Comment by kucint** [comment]
@MartinZikmund 
Unfortunately, I ask you to reopen this issue, because the attached test app is still not working under newest dotnet and newest uno.sdk  "Uno.Sdk": "5.5.43"

**Comment by jeromelaban** [comment]
The change is not presetng in uno 5.5, it will only be in Uno 5.6 at this time.

**Comment by kucint** [comment]
I see, is it present in dev branch?

**Comment by jeromelaban** [comment]
Not yet, we're currently ingesting the .NET 9 changes. It will be available in the dev branch most likely in the next few days.

**Comment by kucint** [comment]
Thanks @jeromelaban for info.

**Comment by agneszitte** [comment]
@jhanvi03, @Darsh0307 this issue will need to be tested with latest internal 5.6 please

**Comment by jhanvi03** [comment]
Both issues are resolved in 5.6 stable versions:
Sdk/template: 5.6.2
VS VSIX: 5.6.5

![Image](https://github.com/user-attachments/assets/e8d22249-c048-4533-aab1-8e8f1c70eb7c)
